app/models/flag.rb

```diff
@@ -7,6 +7,10 @@ class Flag < ActiveRecord::Base
     validates :flaggable_id, presence: true
     validates :flag_type, presence: true
 
+    def self.distinct_task_values(type)
+      return Flag.where(:flag_type => type).pluck(:flag_value).uniq
```

Comment:
What's the sql generated for that? Looks like it's running uniq *after* fetching all rows from db. If that's slow, could be done *in* sql using `DISTINCT` or `GROUP BY`